src/upgrade/connection.rs

```diff
@@ -9,6 +9,12 @@ pub struct RawConnection<Inner> {
     inner: Inner,
```

Comment:
If `Connection` is `RawConnection<Box<dyn T>>` anyways, why isn't `inner just directly `Box<dyn T>`?

Reply:
Good question; not sure. I think we should pick that up when stabilizing the `upgrade`submodule.